hazelcast-client/src/main/java/com/hazelcast/client/config/ClientConfig.java

```diff
@@ -759,4 +760,47 @@ public ClientConfig setUserCodeDeploymentConfig(ClientUserCodeDeploymentConfig u
         this.userCodeDeploymentConfig = userCodeDeploymentConfig;
         return this;
     }
+
+    /**
+     *
+     * @param mapName The name of the map for which the query cache config is to be returned.
+     * @param cacheName The name of the query cache config.
+     * @return The query cache config. If the config does not exist, it is created.
+     */
+    public QueryCacheConfig getOrCreateQueryCacheConfig(String mapName, String cacheName) {
+        Map<String, Map<String, QueryCacheConfig>> allQueryCacheConfig = getQueryCacheConfigs();
+
+        Map<String, QueryCacheConfig> mapQueryCacheConfig = lookupByPattern(allQueryCacheConfig, mapName);
```

Comment:
I'm confused with this name.

It isn't a single config, but a map of configurations.

Is represents all the query cache configs for a map. It will help if the variable would be named to 'queryCacheConfigsForMap' so that the name explains what is happening.